lib/IDE/CodeCompletion.cpp

```diff
@@ -5242,73 +5491,78 @@ class CompletionOverrideLookup : public swift::VisibleDeclConsumer {
-        if (!NameOffset.hasValue())
-          NameOffset = OS.tell();
-      }
-
-      // As for FuncDecl, SubscriptDecl, and VarDecl,
+      // As for FuncDecl, SubscriptDecl, and VarDecl, substitute the result type
+      // with 'OpaqueBaseTy' if specified.
       void printDeclResultTypePre(ValueDecl *VD, TypeLoc &TL) override {
         if (!OpaqueBaseTy.isNull()) {
-          OS << "some ";
+          setNextChunkKind(ChunkKind::Keyword);
+          printText("some ");
+          setNextChunkKind(ChunkKind::Text);
           TL = TypeLoc::withoutLoc(OpaqueBaseTy);
         }
+        CodeCompletionStringPrinter::printDeclResultTypePre(VD, TL);
       }
     };
 
-    llvm::SmallString<256> DeclStr;
-    unsigned NameOffset = 0;
-    {
-      llvm::raw_svector_ostream OS(DeclStr);
-      DeclPrinter Printer(
-          OS, getOpaqueResultType(VD, Reason, dynamicLookupInfo));
-      PrintOptions Options;
-      if (auto transformType = CurrDeclContext->getDeclaredTypeInContext())
-        Options.setBaseType(transformType);
-      Options.SkipUnderscoredKeywords = true;
-      Options.PrintImplicitAttrs = false;
-      Options.ExclusiveAttrList.push_back(TAK_escaping);
-      Options.ExclusiveAttrList.push_back(TAK_autoclosure);
-      Options.PrintOverrideKeyword = false;
-      Options.PrintPropertyAccessors = false;
-      Options.PrintSubscriptAccessors = false;
-      Options.PrintStaticKeyword = !hasStaticOrClass;
-      VD->print(Printer, Options);
-      NameOffset = Printer.NameOffset.getValue();
-    }
+    DeclPrinter Printer(Builder, opaqueResultType);
+    Printer.startPreamble();
+
+    bool modifierAdded = false;
 
-    if (!hasDeclIntroducer && !hasAccessModifier)
-      addAccessControl(VD, Builder);
+    // 'public' if needed.
+    modifierAdded |= addAccessControl(VD, Builder);
 
+    // 'override' if needed
     if (missingOverride(Reason)) {
-      if (!hasDeclIntroducer)
-        Builder.addOverrideKeyword();
-      else {
-        auto dist = Ctx.SourceMgr.getByteDistance(
-            introducerLoc, Ctx.SourceMgr.getCodeCompletionLoc());
-        if (dist <= CodeCompletionResult::MaxNumBytesToErase) {
-          Builder.setNumBytesToErase(dist);
-          Builder.addOverrideKeyword();
-          Builder.addDeclIntroducer(DeclStr.str().substr(0, NameOffset));
-        }
+      Builder.addOverrideKeyword();
+      modifierAdded |= true;
+    }
+
+    // Erase existing introducer (e.g. 'func') if any modifiers are added.
+    if (hasDeclIntroducer && modifierAdded) {
+      auto dist = Ctx.SourceMgr.getByteDistance(
+          introducerLoc, Ctx.SourceMgr.getCodeCompletionLoc());
+      if (dist <= CodeCompletionResult::MaxNumBytesToErase) {
+        Builder.setNumBytesToErase(dist);
+        hasDeclIntroducer = false;
       }
     }
 
-    if (!hasDeclIntroducer && NameOffset != 0)
-      Builder.addDeclIntroducer(DeclStr.str().substr(0, NameOffset));
+    PrintOptions PO;
```

Comment:
Are there any changes in the PrintOptions? It's hard to see from the diff

Reply:
* Added `SkipIntroducerKeywords = hasDeclIntroducer`. Previously, decl introducer was trimmed in post processing printed string.
* Changed `Options.PrintStaticKeyword` to `!hasStaticOrClass` to `!hasStaticOrClass && !hasDeclIntroducer`. Previously if there's a decl introducer, attributes are trimmed with the decl introducer in the post processing
* Added `nonisolated` to `ExclusiveAttrList` because it's required to conform.
* Explicit `PrintSelfAccessKindKeyword = false` but they aren't printed even without it because they are not included in `ExclusiveAttrList`. (They are not needed to be printed because (a) conforming `nonmutating` requirement doesn't need explicit `nonmutating` as it is the default (b) conforming `mutating` requirement is done by implicitly non-mutating member.)